trio/_core/_io_epoll.py

```diff
@@ -201,7 +201,7 @@ class _EpollStatistics:
 class EpollIOManager:
     _epoll: select.epoll = attr.ib(factory=select.epoll)
     # {fd: EpollWaiters}
-    _registered: DefaultDict[int, EpollWaiters] = attr.ib(
+    _registered: defaultdict[int, EpollWaiters] = attr.ib(
```

Comment:
Isn't `defaultdict` a 3.9+ thing? Why is ruff doing this? Cause the `from __future__ import annotations`? or is this a bug.

Reply:
It’s probably that yes. There’s an option to decide whether to preserve the hints making sense at runtime.

Reply:
Yes, it decides what's allowed based on both the version you specify and if a future annotations import exists.

Reply:
There is a setting to make sure runtime typing works, mostly so a specific library can work properly, but in my opinion I agree with ruff's behavior here. Why have outdated versions of the type annotations if we don't have to worry about runtime type annotations?